teuthology/suite/run.py

```diff
@@ -564,39 +569,102 @@ def check_num_jobs(self, jobs_to_schedule):
         if threshold and jobs_to_schedule > threshold:
             util.schedule_fail(msg, dry_run=self.args.dry_run)
 
-    def schedule_suite(self):
+    def descr_to_yamls(self, descriptions, suites_path):
         """
-        Schedule the suite-run. Returns the number of jobs scheduled.
+        Function converts description of the run into sequence of .yaml files.
+        Input "descriptions" is the value of "--descr" parameter;
+        "description" consists of comma-separated list of run description
+        Input:
+        descriptions="rerun/rados/objectstore/{backends/objectstore-bluestore-b supported-random-distro$/{centos_latest}}"
+        suites_path="/cephfs/github.com_ceph_build/qa/suites/"
+        Output:
+        ('rados/objectstore/{backends/objectstore-bluestore-b supported-random-distro$/{ubuntu_latest}}',
+        ['/cephfs/github.com_ceph_build/qa/suites/rados/objectstore/backends/objectstore-bluestore-b.yaml',
+         '/cephfs/github.com_ceph_build/qa/suites/rados/objectstore/supported-random-distro$/ubuntu_latest.yaml'])
```

Comment:
It should be more general, I would suggest to remove this examples from here and make a unittest case for it.

Reply:
`tox -e py3` 
to run only part of it you might try: `tox -e py3 -- teuthology/suite` or even `tox -e py3 -- teuthology/suite/test/test_util.py`

Reply:
or even a separate test: 
```
tox -e py3 -- teuthology/suite/test/test_run_.py::TestRun::test_sha1_exists
```
if you have pytest installed, being in virtualenv enabled similar:
```
pytest teuthology/suite/test/test_run_.py::TestRun::test_sha1_exists
```

Reply:
@kshtsk Thanks, I was able to add unit tests in form of 3 separate, concrete examples. Please take a look.